File: frontend/src/services/InventoryService.js
import api from './api'


export const getInventoryByProductId = async (productId) => {

    const response = await api.get(
        `/api/v1/inventory/product/${productId}`
    )

    return response.data
}


export const updateInventory = async (
    inventoryId,
    inventoryData
) => {

    const response = await api.put(
        `/api/v1/inventory/${inventoryId}`,
        inventoryData
    )

    return response.data
}

File: inventory-service/src/main/java/com/ecommerce/inventory/service/InventoryService.java
package com.ecommerce.inventory.service;

import com.ecommerce.inventory.dto.request.InventoryRequest;
import com.ecommerce.inventory.dto.response.InventoryResponse;
import com.ecommerce.inventory.enums.InventoryStatus;
import org.springframework.data.domain.Page;

public interface InventoryService {

    InventoryResponse createInventory(InventoryRequest request);

    InventoryResponse updateInventory(Long id, InventoryRequest request);

    InventoryResponse getInventoryById(Long id);

    InventoryResponse getInventoryBySku(String sku);

    Page<InventoryResponse> getAllInventory(
            int page,
            int size,
            String sortBy
    );

    Page<InventoryResponse> getInventoryByStatus(
            InventoryStatus status,
            int page,
            int size,
            String sortBy
    );

    void deleteInventory(Long id);
    InventoryResponse getInventoryByProductId(Long productId);
}